### ISSUE TITLE
CollectionWrapper - Set higher index when enableRequests are on

### ISSUE THREAD

**Issue by ricardo-a-alves-alb** [rdkcentral/Lightning-ui#55]
Hello,

Using a List (that uses CollectionWrapper under-the-hood) that has a _lazy-load_ approach  should allow to set a higher index than the current loaded items.

On the following example, there are 50 possible items, however initially it only loads 10 items. However, if we press the <kbd>Enter</kbd> key it tries to set the current index to `15`. However, since it only has 10 items and on the `setIndex` method there's a constraint between 0 and current items length, it restricts the index to 10.

Given the lazy-load logic, it should at least attempt to get more items until we reach the requested index.

[Playground example](https://lightningjs.io/playground/#N4IgJgpgDhB2mwMYEsIGcQC5QAEA2yA5gBYAusyshAVmgPSID2AThFiGaVGpnQ2LAB0aANYBPKAENEIwZABudfETIUqtBiwg4ATIICMABkGGQAGhDKS5SjXoBXZO07de-IaInTZCpQWtqdnSOIAC+FmiIzMhQpBjYIHiS9kjEgrRYoABmyHhsmInJqekYFkyQ7MgAtlAspAAEAIJQUPVZzIxV9QDkgnTNUCXdANwAOrDjTLBoDYyxyIzT9QC89cAzkoQQmGsA7jv6AJw6hmb1xAeGAByn9Yh5kswAwox4LDuGAB6GP78-oaExkhFjN6pIWit6rAILsmi0ABRzUgLaYASiBYEYiHsVTgpEEACNGGAxIJwTB4E9iLkwPDycJSJsIIItqQnpJYPJJGh4aj0eZwNA4AgUOgsABtEB9AYlEAAXXCIBlGQSOTy7GVpRA5XyIGqtWYDQAMiobFQ2h0ut0rKpbBomKwRuN9XU1vUTaDQhbOj0bWago5uuNxhBPgaGpAssk8A17ty0HDWqHSMKEyaArZBAMCIhJMjFmtxvVi-UNsjEPUAPopmpJFO8wuwEvN+qsUj2ZhN4BFlu9gCin0ktYgHtIO27Td7U+LpAk23dyBmZh7097yBrABU5zsAJI15eT1cttBSFBUA4ANgPR97cEkBLyACUIABHezoOI7UjMd-Xm-NtAiFgSQ8B4Rt-2nRZnzfD89wgKowO-d8Vwg0IUOnNDDxbTDm0wlDKymGYf0QUgGwnadSGpNBBErdd4ITVZxUEZjGmYZhJDEeEAFZDFROVBCqcF4XhSszmQMBPlRFYAD56nhYB6mRUg8h2cTPnqUI+RQvCsMrHJmBmRoSOQeQIDI9CS0oxdBEZQh4W6AchygPJR26VFBFYGCZjghDeW04NdOIDkwDyPtYBTZhzKw5srOo2z7Mc4dXPctAIFIHd4FDeF9C41F-ImLCoNfd9vJrHkFLAPNJCNOBCEos5BPUzTwKnNsOybAAFS1F2ZVg0FeUz4Vimi6IQ4QczM+FKsZGqqEo+oAB8FvqRrUTOKaqtmuriEW5bVvqABqeojC0rCdObStWQAMSxexUtpKTyNatL2sUqibM2BLBySxdSDc-KdLjNAEx8+pk1TBcMyoQQXhqRY8Ra4sy2QCtq3g5y8zMx6LOLNrO0R1dWBIr8fwgP8jyYN5mA+T4siyRBDEQBnEHJ1cjXvCA8HHHHV0+HZ4V2KTllk2E6HqHRWf-MR+eIIXZJ2sWJZ56cqkYFIx3qYwuMlqccN1-L8KyW6eWx6LLPeqRSEQYh5LuV53k12m6ayP4NLys6AoulIjexE2CZii282t23KYdr46eZ5m3YNrDWUU9c8ii6c8abAADAASYBhurBOIFCVPdp6Qwgw9gqALS+PlLMrk8HfU2KPenOq8hGvkLNmd3vi7p2YJTm3Js5NIWGpS8hjgDGXLeo492cTKKT5723xgB2Qwx5LZGKzj4gIFNefexT+oeIBgVIApEVUHiSU-UCe0tAFa+7QcJwFUVRkCXiEAQLwMIgA) and pasting the code here just in case:

```js
import Lightning from '@lightningjs/core';
import { List } from '@lightningjs/ui'

export default class App extends Lightning.Application {
    static _template() {
        return {
            ExampleList: {
                type: List,
                itemType: Item,
                spacing: 16,
                enableRequests: true,
                signals: {
                    onRequestItems: true
                }
            }
        }
    }

    _construct() {
        this._items = [...Array(50)].map((_, idx) => ({ title: idx }))
    }

    _firstActive() {
        this.tag('ExampleList').requestItems()
    }

    _handleEnter() {
        this.tag('ExampleList').setIndex(15)
    }

    onRequestItems({ dataLength, max }) {
        return Promise.resolve(this._items.slice((dataLength || max), (dataLength || max) + 10))
    }

    _getFocused() {
        return this.tag('ExampleList')
    }
}

class Item extends Lightning.Component {
    static _template() {
        return {
            rect: true,
            color: 0xffc0cc0c,
            Label: {
                x: (w) => w / 2,
                y: (h) => h / 2,
                mount: 0.5,
            }
        }
    }

    _focus() {
        this.patch({ color: 0xffff0000 })
    }

    _unfocus() {
        this.patch({ color: 0xffc0cc0c })
    }

    get title() {
        return `${this._title}` || '0'
    }

    set title(value) {
        this._title = value
        this.tag('Label').text = this.title
    }

    static get width() {
        return 70
    }

    static get height() {
        return 50
    }
}
```

**Comment by ricardo-a-alves-alb** [comment]
There is also a incoherence in the following object:

https://github.com/rdkcentral/Lightning-ui/blob/bbbd03d2e46ddc72f28ee2500101af1b9760abb5/src/helpers/CollectionWrapper.js#L81-L87

It should be:

```js
if(obj === undefined) {
  obj = {
    previousIndex: 0,
    index: 0,
    dataLength: 0,
    mainIndex: 0,
    previousMainIndex: 0,
    lines: 0
  }
}
```

**Comment by ricardo-a-alves-alb** [comment]
I think we can also benefit from a signal that tells you that new items have been added.

**Comment by jfboeve** [comment]
fixed in version 1.3.8
